fluentd/run.sh

```diff
@@ -123,13 +112,58 @@ if [ "${USE_MUX_CLIENT:-}" = "true" ] ; then
     DIV=`expr $DIV \* 2`
 fi
 
+# Default buffer_type: file if FILE_BUFFER_PATH is specified.
+FILE_BUFFER_PATH=${FILE_BUFFER_PATH:-""}
+if [ "$BUFFER_TYPE" = "file" -a "$FILE_BUFFER_PATH" != "" ] ; then
```

Comment:
BUFFER_TYPE is set in the env section of daemonset and deploymentconfig, respectively...

defaults/main.yml:openshift_logging_fluentd_buffer_type: "file"
templates/fluentd.j2:        - name: "BUFFER_TYPE"
templates/fluentd.j2:          value: "{{ openshift_logging_fluentd_buffer_type | lower }}"

defaults/main.yml:openshift_logging_mux_buffer_type: "file"
templates/mux.j2:        - name: "BUFFER_TYPE"
templates/mux.j2:          value: "{{ openshift_logging_mux_buffer_type | lower }}"

> also, should it be an error if BUFFER_TYPE = file but no FILE_BUFFER_PATH was given?
Right. Currently, the buffer type is silently reset to memory if no FILE_BUFFER_PATH is given.
I'm adding this error.  (Or warning is better?)
echo "ERROR: File is set to buffer_type but FILE_BUFFER_PATH is empty.  Using memory buffer"